### DISCUSSION TITLE
Websockets 5 support?

### DISCUSSION THREAD

**Discussion by datlaunchystark** [josephdadams/TallyArbiter#480] [Ideas]
I'm just curious, is there plans to bring Websockets 5 support to this project since it's build into OBS 28?  The protocol is different in this version and trying to connect causes it to loop and crash both Tally Arbiter and OBS.  Also it now defaults to port 4455 (which can be changed) which Tally Arbiter also uses.

**Comment by JTF4** [comment]
I'll start looking into it.

The only think I don't know about is if we can run WS v4 for tally arbiter and WS v5 for OBS at the same time. @josephdadams any thoughts or insights on this?

One other thing to look at will be any updates they made to the API. If they have not made any updates the the API that break our current process of handing tally data I could probably push out a new version in a few days.

**Reply by hrueger** [reply]
I already had to update a project of mine. Maybe I can clarify:
- There is no new WebSocket Standard, but instead just a new major version of the OBS Websocket API which is used by the OBS Websocket Plugin and the respective Clients (in our case the `obs-websocket-js` lib). That means, we only have to change the OBS lib and the code which uses it. There is no change with our Socket.IO communication whatsoever.
- The OBS Websocket Protocol v5 is vastly different from the v4 one, however, it is quite easy to upgrade. There is a basic upgrade guide here: https://github.com/obs-websocket-community-projects/obs-websocket-js/releases/tag/v5.0.0

Hope that helps :smile:

**Reply by josephdadams** [reply]
There’s also a 4.9.1-compat of ws that can be installed alongside v5 so users can just do that until we fix our end

**Reply by JTF4** [reply]
Do we want to write a quick doc on setting that up?

**Reply by josephdadams** [reply]
Another doc for people to not read

**Reply by TychoWerner** [reply]
In order to get OBS Studio 29.1.3 working you will need to install Websockets v4
See docs: https://josephdadams.github.io/TallyArbiter/docs/usage/sections/sources#obs-studio

**Comment by Jocke4f** [comment]
### Functionality
After PR https://github.com/josephdadams/TallyArbiter/pull/715 should we have the same functionality with websocket 5 as with websocket 4.

### Documenation
The OBS doucmeation has been updated to clarify the webscoket 4 and 5 situation, see https://josephdadams.github.io/TallyArbiter/docs/usage/sections/sources#obs-studio.
